src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -298,8 +298,14 @@ public void removeCommitWebHook(BitbucketWebHook hook) {
 
     @Override
     public void postBuildStatus(BitbucketBuildStatus status) {
-        // TODO use Bitbucket Cloud build status API
-        postCommitComment(status.getHash(), status.getDescription() + ". [See build result](" + status.getUrl() + ")");
+        String path = V2_API_BASE_URL + this.owner + "/" + this.repositoryName + "/commit/" + status.getHash() + "/statuses/build";;
+        try {
+            postRequest(path, serialize(status));
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.log(Level.SEVERE, "Enconding error", e);
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE, "Can not comment on commit", e);
```

Comment:
Perhaps a bit misleading message? `Could not post bitbucket build status` would be more correct message